### PR TITLE
refactor: crm page

### PR DESCRIPTION
**Pull request by yusualhashash**
## How has this been tested?

- [ ] Prod
- [x] Development

## Screenshots (if appropriate):
-branchId:
image
-adding sub company and individual to all pages
image
-add individual table
image
-add edit and delete to action list in sub company and individual
image
-add all translation to crm page
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by a0m0rajab — CHANGES_REQUESTED** [review]

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/[id]/form.tsx [hunk 1]
```diff
@@ -253,17 +342,17 @@ export default function Form({
     }
   }
 
-  async function getSubCompaniesInformation(totalCount = 1000) {
+  async function getSubCompaniesInformationForMerchantx() {
```
[the remark this answers is not in the document]
Comment (yusualhashash) [reply apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/[id]/form.tsx:350]: no it does not implemented yet

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/[id]/form.tsx [hunk 2]
```diff
@@ -204,6 +205,94 @@ export default function Form({
     "fullAddress",
     "typeCode",
   ]);
+
+  const individualSchema = {
```
[the remark this answers is not in the document]
Comment (yusualhashash) [reply apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/[id]/form.tsx:214]: done

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 3]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
     return structuredError(error);
   }
 }
+
+export async function getMerchantsByIdSubMerchants(
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:53]: Can we fix the function name?
```suggestion
export async function getSubMerchantsByMerchantId(
```

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 4]
```diff
@@ -46,3 +47,22 @@ export async function getCrmServiceMerchantsDetailById(
...
+  body: GetApiCrmServiceMerchantsByIdSubMerchantsData,
+) {
```
[the remark this answers is not in the document]
Comment (a0m0rajab) [reply apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:55]: can we apply this too?

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 5]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+  try {
+    const client = await getCRMServiceClient();
+    const response =
+      await client.merchant.getApiCrmServiceMerchantsByIdSubMerchants(body);
+    revalidatePath("/");
+    return {
+      type: "success",
+      data: response,
+      status: 200,
+    } as ServerResponse<Volo_Abp_Application_Dtos_PagedResultDto_16>;
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:66]: we can remove the `as serverResponse from here`

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 6]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+  } catch (error) {
+    return structuredError(error);
+  }
+}
+
+export async function getMerchantsByIdIndivituals(): Promise<
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:72]: typo in the function name.

Comment (a0m0rajab) [reply apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:72]: can we write the name in a better way?

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 7]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+    const response =
+      await client.merchant.deleteApiCrmServiceMerchantsByIdWithComponents(
+        body,
+      );
+    revalidatePath("/");
+    return {
+      type: "success",
+      data: response,
+      status: 200,
+    } as ServerResponse<Volo_Abp_Application_Dtos_PagedResultDto_16>;
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:115]: same as before.

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 8]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+  //     data: response,
+  //     status: 200,
+  //   };
+  //   // as ServerResponse<Volo_Abp_Application_Dtos_PagedResultDto_16>;
+  // } catch (error) {
+  //   return structuredError(error);
+  // }
+}
+
+export async function deleteSubMerchantsByIdWithComponents(
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:100]: let's have a simpler names

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 9]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+      type: "success",
+      data: response,
+      status: 200,
+    } as ServerResponse<Volo_Abp_Application_Dtos_PagedResultDto_16>;
+  } catch (error) {
+    return structuredError(error);
+  }
+}
+
+export async function deleteIndivitualsMerchantsByIdWithComponents(): Promise<
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:121]: lets have a meaningful function name

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts [hunk 10]
```diff
@@ -46,3 +49,101 @@ export async function getCrmServiceMerchantsDetailById(
...
+export async function getMerchantsByIdIndivituals(): Promise<
+  ServerResponse<Individual[]>
+> {
+  // body: GetApiCrmServiceMerchantsByIdSubMerchantsData,
+  "use server";
+  return Promise.resolve({
+    type: "success",
+    data: individualData,
+    status: 200,
+    message: "Indivitual deleted successfully.",
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:81]: let's get the correct message here/

File: apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts (hunk 6 above)
Comment (a0m0rajab) [reply apps/web/src/app/[lang]/app/[type]/crm/actions/merchant.ts:72]: can we use this in the table page as well?

**Review by a0m0rajab — CHANGES_REQUESTED** [review]

**Pull request closed without merging**